### DISCUSSION TITLE
Is it possible to write to a _IO tag?

### DISCUSSION THREAD

**Discussion by aquaman479** [dmroeder/pylogix#227] [Q&A]
Is it possible to write to a _IO_XX_XX_XX tag in a micro850? I was able to write to a created program tag, but when I try to write a tag with a python type of INT to a _IO tag the value does not change (the PLC processor seems to overwrite the tag value).

**Comment by dmroeder** [comment]
Input tag states are conteolled by the voltage present on the terminal. Output tags are conteolled by the program. I'm. Not 100% sure how outputs are handles that are not actually address in the program. If you want to control an output externally, standard practice is to have an internal tag control the output, you can write to an internal tag.

**Comment by aquaman479** [comment]
Thank you for your response. I am trying to write to AI like _IO_X1_AI_00 to simulate a 4-20 mA field device, but every time I write to the tag I am not receiving an error, but the value does not change (might be changing but overwritten by Miro850 processor)

**Reply by TheFern2** [reply]
you cannot change hardware IO, this is controlled on the next plc scan.

**Comment by William-Brumble** [comment]
These statements are true, but I would like to add to it that you can if you inhibit the module, while the scenarios where you'd want to do this are limited.

I work in a QC role and I use pylogix to emulate hardware inputs all the time to test plc logic without having the actual hardware present.
